src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -439,6 +439,40 @@ public async Task<IActionResult> GetCompatibility(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Start a dry run of the deployment, which performs all the steps in the process of performing a deployment except for running 'cdk bootstrap' and 'cdk deploy'.
+        /// This includes, but is not limited to, creating a deployment bundle and generating a CDK project.
+        /// </summary>
+        [HttpPost("session/<sessionId>/dryrun")]
+        [SwaggerOperation(OperationId = "StartDryRunDeployment")]
+        [SwaggerResponse(200, type: typeof(StartDryRunDeploymentOutput))]
+        [Authorize]
+        public async Task<IActionResult> StartDryRunDeployment(string sessionId)
```

Comment:
This doesn't really seem like a dry run operation. It just returns back the path to the CDK project that is generated leaving to testing code to do anything meaningful. If we are going to alter our public API we should make it meaningful and something we can expand on.

This whole operation could to generate the CloudFormation template and return back the template to the user. That can be useful later in the CLI experience as well for users that just want the template and then have the template carried forward in their pipeline to later be deployed. Also the toolkit might want to have a button to view the template before deploying it.

Reply:
Done